### PR TITLE
Resources for notification were created

### PR DESCRIPTION
**Pull request by aliaksandr-volkau**

### CODE REVIEW

File: lib/cirro_io_v2/resources/notification_broadcast.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module CirroIOV2
+  module Resources
+    class NotificationBroadcast < Base
+      ALLOWED_PARAMS = %i[payload recipients notification_channel_id].freeze
```
[the remark this answers is not in the document]
Comment (aliaksandr-volkau) [reply lib/cirro_io_v2/resources/notification_broadcast.rb]: https://github.com/test-IO/cirro-ruby-client/pull/31/commits/d0b0f091f83d076be5ec0e13e3d1e23854efc933

**Review by thebluber — COMMENTED** [review]
would you guys please put .idea related files into gitignore 🙏

File: lib/cirro_io_v2/request_clients/jwt.rb [hunk 2]
```diff
@@ -14,6 +14,7 @@ def initialize(base_url:, private_key:, client_id:)
         @connection = Faraday.new(url: base_url) do |conn|
           conn.request :json
           conn.response :json
+          conn.adapter :net_http
```
Comment (thebluber) [inline lib/cirro_io_v2/request_clients/jwt.rb:17]: 🤔 why net_http?

File: lib/cirro_io_v2/resources/base.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 module CirroIOV2
   module Resources
-    class Base
+    class Base < JsonApiClient::Resource
```
Comment (thebluber) [inline lib/cirro_io_v2/resources/base.rb:5]: API v2 is not based on JsonApi anymore. So you can't use JsonApiClient::Resource gem

File: lib/cirro_io_v2/resources/base.rb [hunk 4]
```diff
@@ -12,6 +12,14 @@ def initialize(client)
       def resource_root
         self.class.name.demodulize.underscore.pluralize
       end
+
+      def params_allowed?(params, allowed)
+        raise 'ParamNotAllowed' if (params.keys - allowed).any?
+      end
+
+      def response_object(response)
+        JSON.parse(response.body.to_json, object_class: OpenStruct)
```
[the remark this answers is not in the document]
Comment (thebluber) [reply lib/cirro_io_v2/resources/base.rb:21]: instead we create custom response classes with struct

File: lib/cirro_io_v2/request_clients/jwt.rb (hunk 2 above)
Comment (aliaksandr-volkau) [reply lib/cirro_io_v2/request_clients/jwt.rb:17]: Without it I'm getting
Screenshot 2022-08-05 at 17 56 50
 error.

Comment (thebluber) [reply lib/cirro_io_v2/request_clients/jwt.rb:17]: is it because of JsonAPI maybe? I don't remember any errors like that. Let me try it locally as well.

Comment (aliaksandr-volkau) [reply lib/cirro_io_v2/request_clients/jwt.rb:17]: I understand, it's strange. Because, as I know, Faraday uses net_http adapter by default.

**Pull request closed without merging**